geoh5py/io/h5_writer.py

```diff
@@ -539,7 +542,11 @@ def write_array_attribute(
             except KeyError:
                 pass
 
+            if isinstance(values, np.ndarray) and np.issubdtype(values.dtype, np.str_):
```

Comment:
same, shouldn't we test the types to ensure eventual error?

Reply:
Guess the previous answer is the same here.

Reply:
Good idea, but will be pretty complex to check since there are so many formats passing through. We would need serious validation mechanics

Reply:
ok, guess letting it as it is